model_compression_toolkit/core/pytorch/reader/graph_builders.py

```diff
@@ -123,17 +101,30 @@ def nodes_builder(model: GraphModule,
             weights.update(named_buffer_weights)
 
         if node.op == GET_ATTR:
-            named_parameters_weights = {CONSTANT: to_numpy(parameter) for name, parameter in
-                                        model.named_parameters() if node.target == name}
-            named_buffer_weights = {CONSTANT: to_numpy(parameter) for name, parameter in
-                                    model.named_buffers() if node.target == name}
-            if len(named_parameters_weights) + len(named_buffer_weights) > 1:
-                raise Exception(
-                    f'Constant parameter can only have one tensor. Here we have {len(named_parameters_weights) + len(named_buffer_weights)}')
-
-            weights.update(named_parameters_weights)
-            weights.update(named_buffer_weights)
-            framework_attr.update(const_size=weights.get(CONSTANT).shape)
+            if node_type == ConstantHolder:
+                named_parameters_weights = {CONSTANT: to_numpy(parameter) for name, parameter in
+                                            model.named_parameters() if node.target == name}
+                named_buffer_weights = {CONSTANT: to_numpy(parameter) for name, parameter in
+                                        model.named_buffers() if node.target == name}
+                if len(named_parameters_weights) + len(named_buffer_weights) > 1:
+                    raise Exception(
+                        f'Constant parameter can only have one tensor. Here we have {len(named_parameters_weights) + len(named_buffer_weights)}')
+
+                weights.update(named_parameters_weights)
+                weights.update(named_buffer_weights)
+                framework_attr.update(const_size=weights.get(CONSTANT).shape)
+            elif node_type == BufferHolder:
+                named_parameters_weights = {BUFFER: to_numpy(parameter) for name, parameter in
+                                            model.named_parameters() if node.target == name}
+                named_buffer_weights = {BUFFER: to_numpy(parameter) for name, parameter in
+                                        model.named_buffers() if node.target == name}
+                if len(named_parameters_weights) + len(named_buffer_weights) > 1:
+                    raise Exception(
+                        f'Constant parameter can only have one tensor. Here we have {len(named_parameters_weights) + len(named_buffer_weights)}')
+
```

Comment:
The code here is duplicated. We can use a function to create a node holder perhaps